pennylane/wires.py

```diff
@@ -25,55 +24,71 @@ class WireError(Exception):
 
 
 def _process(wires):
-    """Converts the input to a tuple of numbers or strings."""
+    """Converts the input to a tuple of wire labels.
 
-    if isinstance(wires, (Number, str)):
-        # interpret as a single wire
-        return (wires,)
+    If `wires` can be iterated over, its elements are interpreted as wire labels
+    and turned into a tuple. Otherwise, `wires` is interpreted as a single wire label.
+
+    The only exception to this are strings, which are always interpreted as a single
+    wire label, so users can address wires with labels such as `"ancilla"`.
 
-    if isinstance(wires, Wires):
-        # if input is already a Wires object, just return its wire tuple
-        return wires.labels
+    Any type can be a wire label, as long as it is hashable. We need this to establish
+    the uniqueness of two labels. For example, `0` and `0.` are interpreted as
+    the same wire label because `hash(0.) == hash(0)` evaluates to true.
+
+    Note that opposed to numpy arrays, `pennylane.numpy` 0-dim array are hashable.
+    """
 
-    if getattr(wires, "shape", None) == tuple():
-        # Scalar NumPy array
-        return (wires.item(),)
+    if isinstance(wires, str):
+        # Interpret string as a non-iterable object.
+        # This is the only exception to the logic
+        # of considering the elements of iterables as wire labels.
+        wires = [wires]
 
-    if isinstance(wires, Iterable):
+    try:
+        # Use tuple conversion as a check for whether `wires` can be iterated over.
+        # Note, this is not the same as `isinstance(wires, Iterable)` which would
+        # pass for 0-dim numpy arrays that cannot be iterated over.
         tuple_of_wires = tuple(wires)
```

Comment:
Some I'm confused about what `tuple` would catch that `set` wouldn't.

Reply:
Oh, it may also be able to decide iterability, but we need a tuple anyways...